### DISCUSSION TITLE
Best practice to instantiate dynamic services (Service Subscribers & Locators)

### DISCUSSION THREAD

**Discussion by michelle-mabelle** [symfony/symfony#43178] [Q&A] [answered]
Hi everyone,

I'm working on _Symfony 5.3/PHP 8_ and i would like to instantiate some (not shared) services dynamically.
Reading the documentation, i've found [Service Subscribers & Locators](https://symfony.com/doc/current/service_container/service_subscribers_locators.html) page that seems to be the best practice.

I wrote a simple example in a [repo Github](https://github.com/michelle-mabelle/symfony-dynamic-service/commit/898963425e0b3fb30ac6243ea703205263f635ab).
Anyway, I am trying to implement this solution but my services are still not found.

What am I missing? Or that I misunderstood?
Thanks for your help.

**Comment by slaci** [comment, marked as answer]
You are using the wrong `ContainerInterface` as the locator. Try this:

```php
use Psr\Container\ContainerInterface;
```

Bonus (not an error)
You don't have to use a key for the service if you don't want to rename it, so you can write:
```php
public static function getSubscribedServices(): array
{
        return [
                ItemA::class,
                ItemB::class,
       ];
}
```

**Reply by michelle-mabelle** [reply]
Shame on me... It works now like a charm !
Thanks a lot slaci !
